### PR TITLE
Implemented support for non-standard installation of oneAPI compiler by dpctl

### LINKED ISSUE #480: Building with oneAPI for non-standard compiler installation
For the default value of `--sycl-compiler-prefix` option to `develop`/`install` setup commands we require `ONEAPI_ROOT` and do not set `DPCTL_CUSTOM_DPCPP_INSTALL_DIR`.

The logic is all built assuming that setting ``DPCTL_CUSTOM_DPCPP_INSTALL_DIR`` signals use of custom-built DPC++ toolchain. 

When oneAPI compiler bits are installed into Python environment, we can not use `ONEAPI_ROOT`, since compiler's location is no longer `${ONEAPI_ROOT}/compiler/latest/linux` but `${CONDA_PREFIX}` instead, and must rely on `--sycl-compiler-prefix` which sets  `DPCTL_CUSTOM_DPCPP_INSTALL_DIR` and CMake script starts assuming that we build with OS DPC++ toolchain.

@AndresGuzman-Ballen

### PR DESCRIPTION
**Pull request by oleksandr-pavlyk**
Closes #480 

@AndresGuzman-Ballen Please check to see if this changes unblocks the use of DPC++ installation in Python environment.

### CODE REVIEW

**Comment by diptorupd** [comment]
@oleksandr-pavlyk I do not like the approach of checking the compiler string. Just checking `dpcpp` is not a good practice. One can use oneAPI and use `-DCMAKE_CXX_COMPILER=clang++ -DCMAKE_CXXFLAGS=-fsycl` and not specify `dpcpp`. Please see my proposed solution in #480.

**Comment by oleksandr-pavlyk** [comment]
> @oleksandr-pavlyk I do not like the approach of checking the compiler string. Just checking `dpcpp` is not a good practice. One can use oneAPI and use `-DCMAKE_CXX_COMPILER=clang++ -DCMAKE_CXXFLAGS=-fsycl` and not specify `dpcpp`. Please see my proposed solution in #480.

Ok, I will implement that, but the only difference would be that we would not insist on 2021.2 minimum version requirement.

**Comment by oleksandr-pavlyk** [comment]
Confirmed that `python setup.py develop` works with default one API, and with custom-toolchain, using ``python setup.py develop --sycl-compiler-prefix=${DPCPP_HOME}/llvm/build``

File: dpctl-capi/CMakeLists.txt [hunk 1]
```diff
@@ -9,10 +9,14 @@ project(
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/cmake/modules/")
 find_package(Git REQUIRED)
 
-option(DPCTL_CUSTOM_DPCPP_INSTALL_DIR
+option(DPCTL_DPCPP_HOME_DIR
     "Use a custom version of DPCPP installed at the provided location."
```
Comment (diptorupd) [inline dpctl-capi/CMakeLists.txt:13]: The message should say "The installation home for the DPCPP compiler."

Comment (oleksandr-pavlyk) [reply dpctl-capi/CMakeLists.txt]: Good catch. Fixed.

File: dpctl-capi/cmake/modules/FindIntelSycl.cmake [hunk 2]
```diff
@@ -49,38 +48,32 @@ elseif(DEFINED ENV{ONEAPI_ROOT})
...
 
 # We will extract the version information from the compiler
-if(USING_ONEAPI_DPCPP)
-    set(dpcpp_cmd "${IntelSycl_ROOT}/bin/dpcpp")
-    set(dpcpp_arg "--version")
-else()
-    set(dpcpp_cmd "${IntelSycl_ROOT}/bin/clang++")
-    set(dpcpp_arg "--version")
-endif()
+set(clangpp_cmd "${IntelSycl_ROOT}/bin/clang++")
```
Comment (diptorupd) [inline dpctl-capi/cmake/modules/FindIntelSycl.cmake:61]: `s\clangpp\clangcpp`

Comment (oleksandr-pavlyk) [reply dpctl-capi/cmake/modules/FindIntelSycl.cmake:61]: clangpp stood for clang++, clangcpp does not :)

Comment (diptorupd) [reply dpctl-capi/cmake/modules/FindIntelSycl.cmake:61]: Aha! then how about `clangxx`

Comment (oleksandr-pavlyk) [reply dpctl-capi/cmake/modules/FindIntelSycl.cmake]: Renamed to `clangxx`.

**Comment by diptorupd** [comment]
LGTM. In @AndresGuzman-Ballen's case all he needs to do is pass in `--sycl_compiler_path=${BUILD_DIR}`. Right?

**Comment by oleksandr-pavlyk** [comment]
Yes, and he already got this working on Linux, and Windows (a test fails here, but building process works)

**Review by diptorupd — APPROVED** [review]
Please squash and merge

**Comment by diptorupd** [comment]
@oleksandr-pavlyk please update the docs in a follow up or as a separate commit in this PR.

**Review by diptorupd — APPROVED** [review]

**Pull request merged**